tools/swift-api-digester/ModuleAnalyzerNodes.cpp

```diff
@@ -1593,7 +1593,21 @@ SDKContext::shouldIgnore(Decl *D, const Decl* Parent) const {
       }
     }
   }
+
   if (checkingABI()) {
+    // Declarations which are unavailable on a specific platform, e.g.
+    // `@available(macOS, unavailable)` but not `@available(*, unavailable)`,
+    // should be ignored if we're checking ABI.
+    if (auto availableAttr = AvailableAttr::isUnavailable(D)) {
```

Comment:
Could you explain the rational here? Why is it OK to exclude platform-specific unavailable decls?

Reply:
This was added to exclude `Float16` (see #32868). The rationale is that a platform-agnostic attribute will be used for legacy ABI.

Reply:
We cannot ignore an unavailable decl if it has a fixed binary order. Could you (1) move this check after all other conditions in this `checkingABI()` block, and (2) add comments explaining why we cannot ignore platform-agnostic  unavailable decls.

Reply:
I wonder if treating `*` specially here has a potential to become a hidden trap later. AIUI, it's supposed to just be a convenient shorthand for separate `@available` attributes for each supported platform, fully interchangeable with its expansion.

Stepping back, do we really want to exclude `Float16` from these checks? libswiftCore.dylib does export symbols for all `Float16` entry points on macOS, so to me it would make sense to raise an alarm if anything changes about them -- even if binaries need to do acrobatics to access them. If we need to change/remove them in the future, it seems easy enough to add them to the right exception list, just like any other ABI-impacting change. (Cc @stephentyrone)

Reply:
@lorentey I think you're right, adding `Float16` ABI stability testing is an improvement. But there aren't separate iOS/tvOS/watchOS baselines, so `Float16` source compatibility won't be tested.

@nkcsgexi The tool change is very simple now, so you might want to close this pull request, and update the baselines yourself (if it's easier to review them locally). I can update LegacyABI.swift in #30780 instead.

Reply:
Sounds good to me!